engine/hooks/hooks_handlers.go

```diff
@@ -2,22 +2,163 @@ package hooks
 
 import (
 	"context"
+	"encoding/json"
 	"errors"
 	"fmt"
 	"io"
 	"net/http"
 	"sort"
 	"strconv"
+	"strings"
 	"time"
 
 	"github.com/gorilla/mux"
 	"github.com/rockbears/log"
 
 	"github.com/ovh/cds/engine/api"
+	"github.com/ovh/cds/engine/cache"
 	"github.com/ovh/cds/engine/service"
 	"github.com/ovh/cds/sdk"
 )
 
+func (s *Service) registerRepositoryHookHandler() service.Handler {
+	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+
+		var newHook sdk.RepositoryWebHook
+		//Read the body
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			return sdk.NewErrorFrom(sdk.ErrWrongRequest, "unable to read request")
+		}
+		if err := json.Unmarshal(body, &newHook); err != nil {
+			return sdk.NewErrorFrom(sdk.ErrInvalidData, "unable to unmarshal request")
+		}
+
+		if len(newHook.Configuration) == 0 {
+			return sdk.NewErrorFrom(sdk.ErrInvalidData, "missing hook configuration")
+		}
+
+		vcsType, has := newHook.Configuration[sdk.HookConfigVCSType]
+		if !has || vcsType.Value == "" {
+			return sdk.NewErrorFrom(sdk.ErrWrongRequest, "missing vcs type")
+		}
+
+		vcsName, has := newHook.Configuration[sdk.HookConfigVCSServer]
+		if !has || vcsName.Value == "" {
+			return sdk.NewErrorFrom(sdk.ErrWrongRequest, "missing vcs name")
+		}
+
+		repoName, has := newHook.Configuration[sdk.HookConfigRepoFullName]
+		if !has || repoName.Value == "" {
+			return sdk.NewErrorFrom(sdk.ErrWrongRequest, "missing repository name")
+		}
+
+		project, has := newHook.Configuration[sdk.HookConfigProject]
+		if !has || project.Value == "" {
+			return sdk.NewErrorFrom(sdk.ErrWrongRequest, "missing project key")
+		}
+
+		if err := s.addTaskFromRepositoryHook(newHook); err != nil {
+			return sdk.WithStack(err)
+		}
+		return nil
+	}
+}
+
+func (s *Service) repositoryHooksHandler() service.Handler {
```

Comment:
s/repository/entity

Reply:
Handler to receive hook from repository, so i think it's ok